dialog/folder.go

```diff
@@ -19,7 +19,10 @@ func NewFolderOpen(callback func(fyne.ListableURI, error), parent fyne.Window) *
 	return dialog
 }
 
-// ShowFolderOpen creates and shows a file dialog allowing the user to choose a folder to open.
+// ShowFolderOpen creates and shows a file dialog allowing the user to choose a
+// folder to open. The callback function will run when the dialog closes.  The
+// URI will be nil when nothing is selected or cancelled.
```

Comment:
The latest change made this slightly ambiguous sorry - "nothing is selected or cancelled".
I think "the user cancelled or nothing is selected" is clearer.